packages/core/test/schema.test.ts

```diff
@@ -53,7 +53,10 @@ describe('schema', () => {
             }
           }
         }),
-        events: createSchema<{ type: 'FOO' } | { type: 'BAR' }>()
```

Comment:
To work around the lack of partial inference I've imagined that we are going to use schema to provide that. This change deviates slightly from it and I'm slightly worried that this will hurt the adoption of this pattern - people are already defining unions for `TEvent`, they just provide that explicitly as a generic to the `createMachine`. I've hoped that they could easily migrate to the schema-based generics and this won't allow that because each event has to be declared separately.

The proposed change also requires `EventFromEventSchemaMap` trickery - which works and is doable but I'm also worried that this will lead to less readable TS errors etc.

Reply:
Yeah, that makes sense.